src/Engine/Utils.java

```diff
@@ -242,8 +259,11 @@ public static Path findShortestPath(Unit unit, int x, int y, GameMap map)
     ArrayList<SearchNode> waypointList = new ArrayList<SearchNode>();
 
     // Find optimal route.
-    while (!searchQueue.isEmpty())
+    int MAX_ITERATIONS = 300; // Put a limit on the number of nodes we can expand.
+    int iters = 0;
+    while (!searchQueue.isEmpty() && (iters < MAX_ITERATIONS))
```

Comment:
Note: Without this limit, I was hitting an `OutOfMemoryError` while searching for paths in Europe. Given that Europe is 104*95 spaces, the theoretical number of search nodes is 9880, but I was seeing Utils with a search queue of over 20 million. Something weird going on.

Reply:
I fixed the issue and removed this limit on a new PR against this branch. We should merge that one, then this one, then `aiFramework`, ideally.